draft-irtf-panrg-path-properties.md

```diff
@@ -124,7 +124,8 @@ Path element:
 Path:
 : A sequence of adjacent path elements over which a packet can be transmitted, starting and ending with a node.
 
-  Paths are unidirectional and time-dependent, i.e., the sequence of path elements over which packets are sent from one node to another may change.
+  Paths are unidirectional and time-dependent, i.e., there can be a variety of paths from one node to another, and the path over which packets are transmitted may change.
+  A path definition can be strict (i.e., the exact sequence of path elements remains the same) or loose (i.e., a subset of path elements may vary over time).
```

Comment:
> fixed start and end endpoint with one or multiple virtual path elements in-between

The fixed part can be any hops in-between. Not sure to see the confusion in the edited text, but I may be missing something.

Reply:
For the loose path definition, I wonder if it makes sense to add:
```suggestion
  A path definition can be strict (i.e., the exact sequence of path elements remains the same) or loose (i.e., the first and last node remain the same, but the path elements between them may vary over time).
```

@boucadair Does this text still conform to your understanding of a loose path definition?
If the start or end change, I would have a hard time considering it the same path.

> We define a path as "a sequence of adjacent path elements".
This means we cannot natively describe a loose path without the use of virtual path elements, since it would be something along the lines of "a sequence of adjacent time-varying path elements", right?

I don't understand why we need virtual path elements here.
Med brought up the loose path definition in the context of Time-Variant Routing ([WG charter](https://datatracker.ietf.org/wg/tvr/about/)). In my mind, a good example is a path over a satellite network, where the actual physical satellites relaying the data can change over the lifetime of a connection, and with a loose path definition, you could still consider it "the same path", even though some path elements changed.
Does that make sense? Maybe I should add this example to the document?

Reply:
> the start or end change

If by first/end path elements, you mean the source/destination, then that's OK. If so, I prefer we used source/destination and not first/last as I interpreted initially those as the first next-hop in front of the src/dst.

Reply:
I suggest using "start and end node" since we also define a path in this way:
> A sequence of adjacent path elements over which a packet can be transmitted, starting and ending with a node.

 > with a loose path definition, you could still consider it "the same path", even though some path elements changed.

I guess that makes sense and we should be open for different ways to define paths.
My main concern was that the rest of the document (which was, at least in my case, written with the strict path definition in mind) will still make sense when using a loose path definition.

I was particularly concerned about the following definitions/properties. But I think the definitions still work with a loose path definition:
- "Flow" should be okay since it can be defined over the complete path, which can be either strict or loose. 
- "Property" is defined over a sequence of path elements, which could be problematic for a loose definition as it will only cover the path at a certain point in time (while it is stable). But we could cover a property of a loose path as an aggregated property over the different (stable) sequences of path elements during different time periods.
- Similarly, "disjointness" and "symmetry" still make sense for a loose path (although a loose path might only be symmetric while it is stable, i.e., not transitioning between different path elements).

Can you think of any other cases where a loose path definition might be problematic?

TLDR I'm okay with including the loose path definition

Reply:
I like "start and end" instead of "first and last", and I changed the text accordingly. I also noticed that our Endpoint definition said "first and last", which I changed to "start and end" so it's all consistent.

With regards to the "Property" definition, in my mind there's multiple ways in which this can play out:
A property can apply to a sequence of path elements at one particular moment, and then if the path elements change, the property either remains the same (e.g., access technology was "satellite network" and still is "satellite network", administrative domain and cost are still the same, path MTU is still the same, ...), or the property changes (e.g., one-way delay of a flow transmitted over the path). We talk about path properties changing over time a lot in the document, I think much of this can apply here. And we do have some language around some properties being more likely to change than others over a certain time frame, but for a loose path, it makes sense that things would change more.
And then, as you say, it's possible to aggregate a path property over all the stable sequences of path elements, for example, all costs, all OWDs (e.g., "the minimum OWD for each stable sequence"), ...
Does that make sense?
Also, thanks for being diligent!

I read the rest of the document and I couldn't find any other cases where the loose path definitions might change things.

Reply:
Indeed, a loose path definition could implement a "Property" in different ways.
Perfect, your suggested changes for the endpoint definition also look good to me.